### PR TITLE
TST: Define sections in pandas/conftest.py

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
part of #31989
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/conftest.py [hunk 1]
```diff
@@ -146,947 +187,962 @@ def ordered_fixture(request):
...
+    ymd = tdf.groupby([lambda x: x.year, lambda x: x.month, lambda x: x.day]).sum()
+    # use Int64Index, to make sure things work
+    ymd.index.set_levels([lev.astype("i8") for lev in ymd.index.levels], inplace=True)
+    ymd.index.set_names(["year", "month", "day"], inplace=True)
+    return ymd
 
 
-@pytest.fixture(params=["nlargest", "nsmallest"])
-def nselect_method(request):
+def _create_multiindex():
```
Comment (SaturnFromTitan) [inline pandas/conftest.py]: renamed from `_gen_mi` and added docstring

**Comment by SaturnFromTitan** [comment]
Apart from one helper function rename and a new docstring (marked with a comment), this is purely a reordering.

I suggest to just read the new `pandas/conftest.py` instead of the "Files changed" tab

**Review by simonjayhawkins — APPROVED** [review]
thanks @SaturnFromTitan lgtm.

**Review by jreback — APPROVED** [review]
lgtm. pls merge master and ping on green.

**Comment by SaturnFromTitan** [comment]
@jreback CI isn't green, but the failure isn't related. The same check fails for a lot of other PRs. -> I think this is good to be merged

**Comment by simonjayhawkins** [comment]
Thanks @SaturnFromTitan

**Pull request merged**